fifteen_min/src/viewer.rs

```diff
@@ -255,6 +255,7 @@ fn options_to_controls(ctx: &mut EventCtx, opts: &Options) -> Widget {
 }
 
 fn options_from_controls(panel: &Panel) -> Options {
+    #![allow(clippy::or_fun_call)]
```

Comment:
https://rust-lang.github.io/rust-clippy/master/#or_fun_call
I don't understand why this would be firing here?

(And I can't easily run clippy locally because my version of Rust doesn't know how to iterate over arrays)

Reply:
It's triggered by `.unwrap_or(WalkingOptions::default_speed())`. I consider this a false positive, because `default_speed` is just a getter method. However, clippy cannot know this. It might simpler to just use `unwrap_or_else`. `unwrap_or` is not really necessary, `unwrap_or_else` should lead to the same code here.

Sorry about the code not running on stable, I was following the clippy suggestions slavishly without considering backwards compatibility. I'll fix that!

Reply:
Thanks for explaining! Yeah, `.unwrap_or_else(WalkingOptions::default_speed)` would be preferable to disabling a lint.

And thanks for fixing the 1.52 array issue, it's building fine now!

Reply:
Ok, I'll change it to `unwrap_or_else`. (I think there were a few other similar places, I'll check them, too.)

Reply:
Done.